File: pset1/mario/less/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int colNum;

    do
    {
        colNum = get_int("Height: ");
    }
    while (colNum < 1 || colNum > 8);

    for (int i = 0; i < colNum; i++)
    {
        for (int j = 0; j < colNum; j++)
        {
            // 0: 0 1 2  ..#
            // 1: 0 1 2. .##
            // 2: 0 1 2  ###

            if (colNum - j - 1 > i)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }
        printf("\n");
    }
}

File: pset6/mario/less/mario.py
# Refactor of pset1/mario/less
from cs50 import get_int


def main():
    # Iterate until user inputs correct value
    while(True):
        cols = get_int("Height: ")

        # Input needs to be between 1-8
        if cols > 0 and cols < 9:
            break

    # Print the pyramid
    print_pyramid(cols)


# Prints right aligned pyramid
def print_pyramid(cols):
    for i in range(1, cols + 1):
        print(" " * (cols - i), end="")
        print("#" * i)


if __name__ == "__main__":
    main()